src/puppetlabs/puppetdb/http/query.clj

```diff
@@ -56,6 +54,36 @@
                    ["=" ["node" "active"] true]]
                   req))
 
+(defn add-extract [projections query]
+  (cm/match [query]
+            [["extract" columns expr :guard nil?]]
```

Comment:
Okay, I have a better answer to all this, but for now I think its best we drop this specialisation and just remove this function completely. I think the extract business here specifically for reports/<hash>/events can wait, the root /v4/events doesn't have it, for now I'm keeping it that way. The long answer, is sorting out the events endpoint properly, but thats going to require more work, in short I think the better answer is better expressiveness and querying for parent data instead of expressing it in the child. I'll talk to you more about it on Monday.